client/js/compute.js

```diff
@@ -0,0 +1,554 @@
+    },
+    {
+      name: 'log2',
+      desc: 'returns the base 2 logarithm of a number',
+      sample: 'var a = log2 ${x}',
+      call: function(v, x) { return v = Math.log2(x) }
+    },
+    {
+      name: 'random',
+      desc: 'returns a decimal number in the range 0 to less than 1',
+      sample: 'var a = random',
+      call: function(v) { return v = Math.random() }
+    },
+    {
+      name: 'round',
+      desc: 'returns the value of a number rounded to the nearest integer',
+      sample: 'var a = round ${x}',
+      call: function(v, x) { return v = Math.round(x) }
+    },
+    {
+      name: 'sign',
+      desc: 'returns 1 or -1, indicating the sign of a number (0 => 0)',
+      sample: 'var a = sign ${x}',
+      call: function(v, x) { return v = Math.sign(x) }
+    },
+    {
+      name: 'sqrt',
+      desc: 'returns the square root of a number',
+      sample: 'var a = sqrt ${x}',
+      call: function(v, x) { return v = Math.sqrt(x) }
+    },
+    {
+      name: 'trunc',
+      desc: 'returns the integer part of a number by removing any digits',
+      sample: 'var a = trunc ${x}',
+      call: function(v, x) { return v = Math.trunc(x) }
+    },
+    {
+      name: 'E',
+      desc: 'represents Euler\'s number, the base of natural logarithms',
+      sample: 'var a = E',
+      call: function(v) { return v = Math.E }
+    },
+    {
+      name: 'LN2',
+      desc: 'represents the natural logarithm of 2',
+      sample: 'var a = LN2',
+      call: function(v) { return v = Math.LN2 }
+    },
+    {
+      name: 'LN10',
+      desc: 'represents the natural logarithm of 10',
+      sample: 'var a = LN10',
+      call: function(v) { return v = Math.LN10 }
+    },
+    {
+      name: 'LOG2E',
+      desc: 'represents the base 2 logarithm of e',
+      sample: 'var a = LOG2E',
+      call: function(v) { return v = Math.LOG2E }
+    },
+    {
+      name: 'LOG10E',
+      desc: 'represents the base 10 logarithm of e',
+      sample: 'var a = LOG10E',
+      call: function(v) { return v = Math.LOG10E }
+    },
+    {
+      name: 'PI',
+      desc: 'represents the ratio of the circumference of a circle to its diameter',
+      sample: 'var a = PI',
+      call: function(v) { return v = Math.PI }
+    },
+    {
+      name: 'SQRT1_2',
+      desc: 'represents the square root of 1/2',
+      sample: 'var a = SQRT1_2',
+      call: function(v) { return v = Math.SQRT1_2 }
+    },
+    {
+      name: 'SQRT2',
+      desc: 'represents the square root of 2',
+      sample: 'var a = SQRT2',
+      call: function(v) { return v = Math.SQRT2 }
+    },
+    {
+      name: 'length',
+      desc: 'returns the length of an array',
+      sample: 'var a = ${x} length',
+      call: function(v, x) { return v = x.length }
+    },
+    {
+      name: 'parseFloat',
+      desc: 'returns a floating point number parsed from the given string',
+      sample: 'var a = parseFloat x',
+      call: function(v, x) { return v = parseFloat(x) }
+    },
```

Comment:
Line 299 should be:
sample: 'var a = parseFloat ${x}',
(curly brackets missing)